File: src/codes/exposed-complexity/py/uncode.py
class AuthService:
    def authenticate(self, user_id: str) -> bool:
        print(f"ユーザー認証: {user_id}")
        return True

class UserProfileService:
    def load(self, user_id: str) -> dict:
        print(f"プロフィール取得: {user_id}")
        return {"name": "Hiroshi", "id": user_id}

class AuditService:
    def log(self, action: str):
        print(f"[監査ログ] {action}")

# 呼び出し元がすべて処理を組み立てている
def show_user_profile(user_id: str):
    auth = AuthService()
    profile_service = UserProfileService()
    audit = AuditService()

    if auth.authenticate(user_id):
        profile = profile_service.load(user_id)
        print(profile)
        audit.log("プロフィール表示")

# 実行
show_user_profile("user-123")

File: src/codes/exposed-complexity/ts/uncode.ts
class AuthService {
  authenticate(userId: string): boolean {
    console.log(`ユーザー認証: ${userId}`);
    return true;
  }
}

class UserProfileService {
  load(userId: string): object {
    console.log(`プロフィール取得: ${userId}`);
    return { name: "Hiroshi", id: userId };
  }
}

class AuditService {
  log(action: string) {
    console.log(`[監査ログ] ${action}`);
  }
}

// 呼び出し元が全部自分で処理を組み立てている
function showUserProfile(userId: string) {
  const auth = new AuthService();
  const profileService = new UserProfileService();
  const audit = new AuditService();

  if (auth.authenticate(userId)) {
    const profile = profileService.load(userId);
    console.log(profile);
    audit.log("プロフィール表示");
  }
}
